cmd/root.go

```diff
@@ -222,6 +232,35 @@ func preExecHook(cmd *cobra.Command, args []string) {
 			log.Fatalf("fsoc is not configured, please use \"fsoc config set\" to configure an initial context")
 		}
 	}
+
+	// Do version checking
+	noUpdateWarning, _ := cmd.Flags().GetBool("no-update-warning")
+	updateChecked := int(time.Now().Unix())-getRecentTimestamp() > int(secondInDay) && !noUpdateWarning
+	updateChannel = make(chan *semver.Version)
+	if updateChecked {
+		go version.CheckForUpdate(updateChannel)
+	} else {
+		go func() { updateChannel <- version.ConvertVerToSemVar(version.GetVersion()) }()
```

Comment:
If the purpose of this is just to not do (skip) the check, isn't there a better way?